wsagent/che-core-api-languageserver-shared/src/main/java/org/eclipse/che/api/languageserver/shared/model/LanguageDescription.java

```diff
@@ -62,16 +64,22 @@ public void setMimeType(final String mimeType) {
         return this.fileExtensions;
     }
 
+    /**
+     * @param fileExtensions must not be null
+     */
     public void setFileExtensions(final List<String> fileExtensions) {
-        this.fileExtensions = fileExtensions;
+        this.fileExtensions = new ArrayList<>(fileExtensions);
```

Comment:
it seems per the test result that we should do the assignment only if fileExtensions is not null (or use another pattern)